The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/12224274723/job/34096830447) [(plain)](https://github.com/rust-lang-ci/rust/commit/a41be64cd98c9df1e78863da4c6a1da8d9de2038/checks/34096830447/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [crashes] tests\crashes\132335.rs ... ok
test [crashes] tests\crashes\132430.rs ... ok
test [crashes] tests\crashes\132766.rs ... ok
test [crashes] tests\crashes\132765.rs ... ok
2024-12-08T19:28:29.926514Z ERROR compiletest::runtest: fatal error, panic: "crashtest no longer crashes/triggers ICE, horray! Please give it a meaningful name, add a doc-comment to the start of the test explaining why it exists and move it to tests/ui or wherever you see fit. Adding 'Fixes #<issueNr>' to your PR description ensures that the corresponding ticket is auto-closed upon merge. If you want to see verbose output, set `COMPILETEST_VERBOSE_CRASHES=1`."
test [crashes] tests\crashes\132882.rs ... ok
test [crashes] tests\crashes\133063.rs ... ok
test [crashes] tests\crashes\133066.rs ... ok
test [crashes] tests\crashes\133275-1.rs ... ok
---
failures:

---- [crashes] tests\crashes\132981.rs stdout ----

error: crashtest no longer crashes/triggers ICE, horray! Please give it a meaningful name, add a doc-comment to the start of the test explaining why it exists and move it to tests/ui or wherever you see fit. Adding 'Fixes #<issueNr>' to your PR description ensures that the corresponding ticket is auto-closed upon merge. If you want to see verbose output, set `COMPILETEST_VERBOSE_CRASHES=1`.
thread '[crashes] tests\crashes\132981.rs' panicked at src\tools\compiletest\src\runtest\crashes.rs:17:18:
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
failures:
    [crashes] tests\crashes\132981.rs

test result: FAILED. 212 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 7.81s

Some tests failed in compiletest suite=crashes mode=crashes host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
Build completed unsuccessfully in 0:46:39
make: *** [Makefile:106: ci-msvc-ps1] Error 1
  network time: Sun, 08 Dec 2024 19:28:34 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

--- a/tests/crashes/132765.rs
+++ b/tests/crashes/132765.rs
@@ -1,0 +1,12 @@
+//@ known-bug: #132765
+
+trait LendingIterator {
+    type Item<'q>;
+    fn for_each(&self, _f: Box<fn(Self::Item<'_>)>) {}
+}
+
+fn f(_: ()) {}
+
+fn main() {
+    LendingIterator::for_each(&(), f);
+}

--- a/tests/crashes/132766.rs
+++ b/tests/crashes/132766.rs
@@ -1,0 +1,9 @@
+//@ known-bug: #132766
+
+trait Trait {}
+impl<'a> Trait for () {
+    fn pass2<'a>() -> impl Trait2 {}
+}
+
+trait Trait2 {}
+impl Trait2 for () {}

--- a/tests/crashes/132882.rs
+++ b/tests/crashes/132882.rs
@@ -1,0 +1,13 @@
+//@ known-bug: #132882
+
+use std::ops::Add;
+
+pub trait Numoid
+where
+    for<N: Numoid> &'a Self: Add<Self>,
+{
+}
+
+pub fn compute<N: Numoid>(a: N) -> N {
+    &a + a
+}

--- a/tests/crashes/132981.rs
+++ b/tests/crashes/132981.rs
@@ -1,0 +1,7 @@
+//@ known-bug: #132981
+//@compile-flags: -Clink-dead-code=true --crate-type lib
+//@ only-x86_64
+//@ ignore-windows
+
+#![feature(rust_cold_cc)]
+pub extern "rust-cold" fn foo(_: [usize; 3]) {}

--- a/tests/crashes/133063.rs
+++ b/tests/crashes/133063.rs
@@ -1,0 +1,8 @@
+//@ known-bug: #133063
+
+fn foo(x: !) {
+    match x {
+        (! | !) if false => {}
+        _ => {}
+    }
+}

--- a/tests/crashes/133066.rs
+++ b/tests/crashes/133066.rs
@@ -1,0 +1,12 @@
+//@ known-bug: #133066
+trait Owner {
+    const C<const N: u32>: u32;
+}
+
+impl Owner for () {;}
+
+fn take0<const N: u64>(_: impl Owner<C<N> = { N }>) {}
+
+fn main() {
+    take0::<f32, >(());
+}

--- a/tests/crashes/133199.rs
+++ b/tests/crashes/133199.rs
@@ -1,0 +1,11 @@
+//@ known-bug: #133199
+//@ aux-build: aux133199.rs
+
+extern crate aux133199;
+
+use aux133199::FixedBitSet;
+
+fn main() {
+    FixedBitSet::<7>::new();
+    //~^ ERROR
+}

--- a/tests/crashes/133275-1.rs
+++ b/tests/crashes/133275-1.rs
@@ -1,0 +1,13 @@
+//@ known-bug: #133275
+#![feature(const_trait_impl)]
+#![feature(associated_type_defaults)]
+
+#[const_trait]
+trait Foo3<T>
+where
+    Self::Baz: Clone,
+{
+    type Baz = T;
+}
+
+pub fn main() {}

--- a/tests/crashes/133275-2.rs
+++ b/tests/crashes/133275-2.rs
@@ -1,0 +1,15 @@
+//@ known-bug: #133275
+#![feature(const_trait_impl)]
+#[const_trait]
+pub trait Owo<X = <IntEnum as Uwu>::T> {}
+
+#[const_trait]
+trait Foo3<T>
+where
+    Self::Bar: Clone,
+    Self::Baz: Clone,
+{
+    type Bar = Vec<Self::Baz>;
+    type Baz = T;
+    //~^ ERROR the trait bound `T: Clone` is not satisfied
+}

--- a/tests/crashes/auxiliary/aux133199.rs
+++ b/tests/crashes/auxiliary/aux133199.rs
@@ -1,0 +1,13 @@
+#![allow(incomplete_features)]
+#![feature(generic_const_exprs)]
+
+pub struct FixedBitSet<const N: usize>;
+
+impl<const N: usize> FixedBitSet<N>
+where
+    [u8; N.div_ceil(8)]: Sized,
+{
+    pub fn new() -> Self {
+        todo!()
+    }
+}

